server/channels/app/shared_channel.go

```diff
@@ -87,15 +105,103 @@ func (a *App) GetSharedChannelsCount(opts model.SharedChannelFilterOpts) (int64,
 }
 
 func (a *App) UpdateSharedChannel(sc *model.SharedChannel) (*model.SharedChannel, error) {
-	return a.Srv().Store().SharedChannel().Update(sc)
+	scUpdated, err := a.Srv().Store().SharedChannel().Update(sc)
+	if err != nil {
+		return nil, err
+	}
+	a.notifyClientsForSharedChannelUpdate(scUpdated)
+	return scUpdated, nil
 }
 
-func (a *App) DeleteSharedChannel(channelID string) (bool, error) {
-	return a.Srv().Store().SharedChannel().Delete(channelID)
+func (a *App) UnshareChannel(channelID string) (bool, error) {
+	// fetch the SharedChannel first
+	sc, err := a.GetSharedChannel(channelID)
+	if err != nil {
+		return false, err
+	}
+
+	// deletes the ShareChannel, unsets the share flag on the channel, deletes all remotes for the channel
+	deleted, err := a.Srv().Store().SharedChannel().Delete(channelID)
+	if err != nil {
+		return false, err
+	}
+	a.notifyClientsForSharedChannelUpdate(sc)
```

Comment:
Is this notifying the client with the old data? Shouldn't it be notifying the client somehow that this has been deleted?

Reply:
Thanks @jespino.  It's actually just using the channel ID from that struct.  I think I'll change the signature to reflect that.  It does raise another good point... the ws event being used is `WebsocketEventChannelConverted` which is normally sent when a channel is converted between public and private.  That is likely  not be the best event to use for deletion.  I'll look into that.

Reply:
Ah, looks like `WebsocketEventChannelConverted` is the right ws event.  We're converting from shared to unshared, and need the client to refresh the channel icon.

Reply:
Updated `notifyClientsForSharedChannelUpdate` to just take a team ID and channel ID to avoid confusion.